.github/workflows/ci.yaml

```diff
@@ -7,64 +7,120 @@ on:
-      - run: sudo apt-get install -y musl-tools
-      - run: make test-all
+
+      - name: Install MUSL
+        run: sudo apt-get install -y musl-tools
+
+      - name: Cargo Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            parser/target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build Parser
+        run: make parser_cli
+
+      - uses: actions/upload-artifact@v1
+        with:
+          name: parser_binary
+          path: parser/target/x86_64-unknown-linux-musl/release/parser
+
+  build_connectors:
+    needs: build_parser
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        connector:
+          - source-gcs
+          - source-hello-world
+          - source-kinesis
+          - source-s3
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Prepare
+        id: prep
+        run: |
+          TAG=$(echo $GITHUB_SHA | head -c7)
+          echo ::set-output name=tag::${TAG}
+
+      - uses: actions/download-artifact@v1
+        with:
+          name: parser_binary
+          path: parser/target/x86_64-unknown-linux-musl/release/
+
+      # Downloaded artifacts lose their prior permission settings
+      - name: Adjust parser permissions
+        run: chmod +x parser/target/x86_64-unknown-linux-musl/release/parser
+
       - name: Set up Cloud SDK
+        if: matrix.connector == 'source-gcs'
         uses: google-github-actions/setup-gcloud@master
         with:
           project_id: ${{ secrets.GCP_PROJECT_ID }}
           service_account_key: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}
           export_default_credentials: true
+
       - name: Configure AWS credentials from Test account
+        if: matrix.connector == 'source-kinesis' || matrix.connector == 'source-s3'
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-2
-      - name: Connector integration tests
-        run: make integration-tests
+
+      - name: Login to GitHub package docker registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | \
+            docker login --username ${{ github.actor }} --password-stdin ghcr.io
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build ${{ matrix.connector }} Docker Image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ${{ matrix.connector }}/Dockerfile
+          load: true
+          tags: ghcr.io/estuary/${{ matrix.connector }}:test
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
`cache-to`/`cache-from` are using the built-in GitHub Actions Cache API.

`mode=max` tells the cache to also cache layers from intermediate builds of a multi-stage Dockerfile. This allows us to cache the builder stage, and thus, all the specific library dependencies of each connector.